### PR TITLE
Addition of all Terraform sample code.

### PR DESCRIPTION
**Pull request by jingiris05**

### CODE REVIEW

File: policies/ucs_server_policies/management/imc_access_policy/terraform/imc_access_policy.tf [hunk 1]
```diff
@@ -1,27 +1,61 @@
...
   }
+
   organization {
     object_type = "organization.Organization"
     moid        = var.organization
   }
+
   configuration_type {
-    configure_inband = true
+    configure_inband      = true
```
Comment (Ghufz) [inline policies/ucs_server_policies/management/imc_access_policy/terraform/imc_access_policy.tf:46]: remove the tab

File: policies/ucs_server_policies/management/imc_access_policy/terraform/imc_access_policy.tf [hunk 2]
```diff
@@ -1,27 +1,61 @@
...
     configure_out_of_band = false
-    object_type = "access.ConfigurationType"
+    object_type           = "access.ConfigurationType"
```
Comment (Ghufz) [inline policies/ucs_server_policies/management/imc_access_policy/terraform/imc_access_policy.tf:48]: remove extra tab

File: policies/ucs_server_policies/misc/ssh_policy/terraform/ssh_policy.tf [hunk 3]
```diff
@@ -1,3 +1,22 @@
+provider "intersight" {
+  endpoint        = "https://intersight.com"
+  apikey          = "xxxxx27564612d30dxxxxx/5f21c9d97564612d30dd575a/5f9a8b877564612xxxxxxxx"
+  secretkey       = "C:\\secretKey.txt"
+}
+
+resource "intersight_server_profile" "server1" {
```
Comment (Ghufz) [inline policies/ucs_server_policies/misc/ssh_policy/terraform/ssh_policy.tf:7]: remove this server profile creation and reference in ssh policy

File: policies/ucs_server_policies/network/san_connectivity_policy/terraform/san_connectivity_policy.tf [hunk 4]
```diff
@@ -1,20 +1,36 @@
+provider "intersight" {
+  endpoint        = "https://intersight.com"
+  apikey          = "xxxxx27564612d30dxxxxx/5f21c9d97564612d30dd575a/5f9a8b877564612xxxxxxxx"
+  secretkey       = "C:\\secretKey.txt"
+}
+
+resource "intersight_server_profile" "server1" {
```
Comment (Ghufz) [inline policies/ucs_server_policies/network/san_connectivity_policy/terraform/san_connectivity_policy.tf:7]: do not create server profile in policy sample, it is already created as part of profile, remove it

File: profile/ucs_domain_profile/terraform/ucs_domain_profile.tf [hunk 5]
```diff
@@ -1,10 +1,238 @@
...
+    },
+    {
+      additional_properties = ""
+      class_id              = "fabric.PortPolicy"
+      object_type           = "fabric.PortPolicy"
+      moid                  = intersight_fabric_port_policy.fabric_port_policy1.moid
+      selector              = ""
+    },
+    {
+      additional_properties = ""
```
Comment (Ghufz) [inline profile/ucs_domain_profile/terraform/ucs_domain_profile.tf:145]: remove all the properties which has empty value

File: profile/ucs_server_profile/terraform/ucs_server_profile.tf [hunk 6]
```diff
@@ -1,17 +1,148 @@
...
+    object_type = "SnmpTrap"
+  }
+
+  organization {
+    object_type = "organization.Organization"
+    moid        = var.organization
+  }
+}
+
+# Create a server profile template
```
Comment (Ghufz) [inline profile/ucs_server_profile/terraform/ucs_server_profile.tf:73]: what is the use of server profile template creation here ?

**Review by Ghufz — CHANGES_REQUESTED** [review]

File: profile/ucs_domain_profile/terraform/ucs_domain_profile.tf (hunk 5 above)
Comment (jingiris05) [reply profile/ucs_domain_profile/terraform/ucs_domain_profile.tf:145]: Both attributes "additional_properties" and "selector" are required.

File: profile/ucs_chassis_profile/terraform/ucs_chassis_profile.tf [hunk 7]
```diff
@@ -1,16 +1,165 @@
...
+      moid        = intersight_access_policy.access1.moid
+      class_id    = "policy.AbstractPolicy"
+      additional_properties = ""
+      selector = ""
+    },
+    {
+      object_type = "snmp.Policy"
+      moid        = intersight_snmp_policy.snmp1.moid
+      class_id    = "policy.AbstractPolicy"
+      additional_properties = ""
```
[the remark this answers is not in the document]
Comment (jingiris05) [reply profile/ucs_chassis_profile/terraform/ucs_chassis_profile.tf:127]: Both attributes "additional_properties" and "selector" are required.

**Review by Ghufz — CHANGES_REQUESTED** [review]

File: policies/ucs_domain_policies/syslog_policy/terraform/syslog_policy.tf [hunk 8]
```diff
@@ -21,22 +39,14 @@ resource "intersight_syslog_policy" "syslog1" {
...
-   type = string
-   description = "<value for organization>"
- }
-
- variable "profile"{
-   type = string
-   description = "Moid of server.Profile"
- }
\ No newline at end of file
+variable "profile" {
```
Comment (Ghufz) [inline policies/ucs_domain_policies/syslog_policy/terraform/syslog_policy.tf:49]: remove profile here, it is not required we are creating here syslog_policy

File: profile/ucs_chassis_profile/terraform/ucs_chassis_profile.tf [hunk 9]
```diff
@@ -1,16 +1,163 @@
...
+  snmp_traps {
+    destination = "10.10.10.1"
+    enabled     = false
+    port        = 660
+    type        = "Trap"
+    user        = "demouser"
+    nr_version  = "V3"
+    object_type = "snmp.Trap"
+  }
+  profiles {
```
Comment (Ghufz) [inline profile/ucs_chassis_profile/terraform/ucs_chassis_profile.tf:79]: why profile is being set here?

File: profile/ucs_chassis_profile/terraform/ucs_chassis_profile.tf [hunk 10]
```diff
@@ -1,16 +1,163 @@
...
+  type        = string
+  description = "value for auth_password"
+}
+
+variable "privacy_password" {
+  type        = string
+  description = "value for privacy password"
+}
+
+variable "profile" {
```
Comment (Ghufz) [inline profile/ucs_chassis_profile/terraform/ucs_chassis_profile.tf:160]: why we are declaring variable for profile

**Review by Ghufz — CHANGES_REQUESTED** [review]
when you address the review comment please mark it resolved, if it is not require any changes please provide the comment for that

File: policies/ucs_server_policies/management/imc_access_policy/terraform/imc_access_policy.tf (hunk 1 above)
Comment (jingiris05) [reply policies/ucs_server_policies/management/imc_access_policy/terraform/imc_access_policy.tf]: Removed.

File: policies/ucs_server_policies/misc/ssh_policy/terraform/ssh_policy.tf (hunk 3 above)
Comment (jingiris05) [reply policies/ucs_server_policies/misc/ssh_policy/terraform/ssh_policy.tf]: Removed and referenced.

File: policies/ucs_server_policies/network/san_connectivity_policy/terraform/san_connectivity_policy.tf (hunk 4 above)
Comment (jingiris05) [reply policies/ucs_server_policies/network/san_connectivity_policy/terraform/san_connectivity_policy.tf]: Removed.

File: profile/ucs_server_profile/terraform/ucs_server_profile.tf (hunk 6 above)
Comment (jingiris05) [reply profile/ucs_server_profile/terraform/ucs_server_profile.tf]: Removed and addressed.

File: policies/ucs_domain_policies/syslog_policy/terraform/syslog_policy.tf (hunk 8 above)
Comment (jingiris05) [reply policies/ucs_domain_policies/syslog_policy/terraform/syslog_policy.tf:49]: Removed.

File: profile/ucs_chassis_profile/terraform/ucs_chassis_profile.tf (hunk 9 above)
Comment (jingiris05) [reply profile/ucs_chassis_profile/terraform/ucs_chassis_profile.tf]: Addressed

File: profile/ucs_chassis_profile/terraform/ucs_chassis_profile.tf (hunk 10 above)
Comment (jingiris05) [reply profile/ucs_chassis_profile/terraform/ucs_chassis_profile.tf]: Removed .

File: template/ucs_server_profile_template/terraform/ucs_server_profile_template.tf [hunk 11]
```diff
@@ -1,42 +1,69 @@
...
+  organization {
+    object_type = "organization.Organization"
+    moid = data.intersight_organization_organization.organization.id
+  }
+
+  policy_bucket {
+    moid        = intersight_ntp_policy.ntp_policy.moid
+    object_type = "ntp.Policy"
+    class_id    = "policy.AbstractPolicy"
+    selector    = ""
```
Comment (Ghufz) [inline template/ucs_server_profile_template/terraform/ucs_server_profile_template.tf:53]: does it still require to set selector as empty ""

**Review by Ghufz — CHANGES_REQUESTED** [review]

**Review by Ghufz — APPROVED** [review]
Looks good me

**Pull request merged**